### PR TITLE
KAFKA-4374 : Update log message for clarity metadata response errors

### PR DESCRIPTION
**Pull request by Nancy-ksolves**
Update the log message in NetworkClient.java for metadata response errors.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/NetworkClient.java [hunk 1]
```diff
@@ -1210,7 +1210,7 @@ public void handleSuccessfulResponse(RequestHeader requestHeader, long now, Meta
             // Check if any topic's metadata failed to get updated
             Map<String, Errors> errors = response.errors();
             if (!errors.isEmpty())
-                log.warn("Error while fetching metadata with correlation id {} : {}", requestHeader.correlationId(), errors);
+                log.warn("The metadata response from the cluster reported a recoverable issue with correlation id {} : {}", requestHeader.correlationId(), response.errors());
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/clients/NetworkClient.java]: Please reuse `errors` to avoid creating collection again.

**Review by chia7712 — COMMENTED** [review]
@Nancy-ksolves thanks for this patch

File: clients/src/main/java/org/apache/kafka/clients/NetworkClient.java (hunk 1 above)
Comment (Nancy-ksolves) [reply clients/src/main/java/org/apache/kafka/clients/NetworkClient.java]: @chia7712 - Mentioned changes has been done. Kindly review it.

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by chia7712** [comment]
@Nancy-ksolves could you please rebase code to trigger QA again?

**Comment by Nancy-ksolves** [comment]
@chia7712 , code rebase has been done.

**Review by chia7712 — APPROVED** [review]

**Pull request merged**